site-src/geps/gep-1282.md

```diff
@@ -0,0 +1,107 @@
+  * Istio uses a [DestinationRule resource with ClientTLSSettings](https://istio.io/latest/docs/reference/config/networking/destination-rule/#ClientTLSSettings) to capture TLS details, and the DestinationRule resource also holds traffic policy information like load balancing algorithm, connection pool size, and so on.
+  * Openshift’s Route resource allows the [configuration of reencryption](https://docs.openshift.com/container-platform/4.10/networking/routes/secured-routes.html#nw-ingress-creating-a-reencrypt-route-with-a-custom-certificate_secured-routes) specifically, along with custom certificate details.
+  * Contour’s HTTPProxy captures TLS details using an Envoy client certificate, destination CA certificate, and optional SubjectName which sets what Envoy should expect to see from the backend service, all inside the HTTPProxy resource. It also requires either a Protocol field inside the HTTProxy, or an annotation on the Service that tells Contour that the service expects TLS. This is [all documented](https://projectcontour.io/docs/v1.21.1/config/upstream-tls/), but I should note that Contour’s docs use the Envoy convention where a backend in Gateway parlance is called an Upstream (which may be confusing if you’re not used to it).
+  * Linkerd uses a [Server resource](https://linkerd.io/2.11/reference/authorization-policy/#server) (which is functionally pretty similar to Service in that it associates a name with a Pod selector, but also has other details like if the service supports Proxy protocol), along with a ServerAuthorization resource that specifies some constructs that sit more at the service mesh level, including identity and access control.
+
+In terms of other implementations existing use cases for features like this:
+- For Contour, there are annotations to allow the configuration of the following on a backend Service:
+  - Max connections
+  - Max pending requests
+  - Max requests
+  - Max retries
+  - Upstream protocol: This is what allows Contour to handle switching protocols for the backend service.
+- In Istio, the `DestinationRule` resource allows the configuration of many settings like this:
+  - Load balancer algorithms
+  - Connection pool settings
+  - Outlier Detection
+  - Tunnel settings for non-HTTP workloads
+- In Consul, [`ServiceDefaults`](https://www.consul.io/docs/connect/config-entries/service-defaults) allows specifying similar configuration:
+  - Max connections
+  - Max pending requests
+  - Max concurrent requests
+  - Protocol
+  - Passive health check interval and max failures (outlier detection)
+  - TLS SNI (we don't support configuring a specific CA at this granularity, but publishing a public key in a status field could be useful)
+  
+The properties we're talking about all share two things:
+- They are tightly bound to the backend, rather than being important at the route level
```

Comment:
Maybe related to the caller as well

Reply:
They're relevant to the caller in how they take effect, but they are (or should be) _owned_ by the application owner, and apply to the service's pods and their associated endpoints, in Gateway API resource terms anyway.

Reply:
Not exactly, in some case, A caller may use LB algorithm and connection pool settings different from caller B. How should gateway API allow this settings